repository/managers.py

```diff
@@ -137,33 +138,32 @@ def item(self, groupId, itemId):
                 'url': item_data.get('item', {}).get('url')
             }
 
-            # Extract Giles upload details if available
             giles_uploads = item_data.get('item', {}).get('gilesUploads', [])
 
-            if giles_uploads:
-                giles_details = []
-                extracted_text = giles_uploads[0].get('extractedText', {})
-
-                if extracted_text and extracted_text.get('content-type') == 'text/plain':
-                    extracted_text_data = get_giles_document_details(self.user, extracted_text.get('id'))
-                    item_data['item']['text'] = extracted_text_data
-                elif giles_uploads[0].get('pages'):
-                    pages = giles_uploads[0].get('pages')
-                    text = ""
-                    for page in pages:
-                        if page.get('text') and page.get('text').get('content-type') == 'text/plain':
-                            data = get_giles_document_details(self.user, page.get('text').get('id'))
-                            text += data
-                    item_data['item']['text'] = text
-                else:
-                    item_data['item']['text'] = "No valid text/plain content found."
+            if not giles_uploads:
+                raise GilesUploadError("No Giles uploads available for this item.")
+            
+            giles_details = []
+            extracted_text = giles_uploads[0].get('extractedText', {})
+
+            if extracted_text and extracted_text.get('content-type') == 'text/plain':
+                extracted_text_data = get_giles_document_details(self.user, extracted_text.get('id'), repository)
+                item_data['item']['text'] = extracted_text_data
+            elif giles_uploads[0].get('pages'):
```

Comment:
hm, I don't think we want this. if there are individual pages, then the count would be off when they get appended together. I think for now, we ignore this. If there is no extracted text, then we just can't annotate the text. However, if a text file has been submitted to Giles, I don't think the text comes back as extracted text but as uploaded file. Have you check that?

Reply:
in that case, we want to annotate the uploaded text.

Reply:
When I uploaded a file, I did get `extractedText` back along with `uploadedFile`, so can I keep using extractedText to get the text?

```
  "uploadedFile": {
      "@type": "GilesFile",
      "id": "",
      "filename": "abcd.pdf",
      "url": "",
      "size": ,
      "processor": ,
      "content-type": "application/pdf"
  },
  "extractedText": {
      "@type": "GilesFile",
      "id": "",
      "filename": "abcd.pdf.txt",
      "url": "",
      "size": 6681,
      "processor": null,
      "content-type": "text/plain"
}
```